section2/methods.rb

```diff
@@ -23,35 +19,56 @@ def add(num2, num1)
 end
 
 def modulus(num1, num2)
-  if num2.to_f.eql?(0.0)
-    return "invalid, as #{num1} could not be modulus by #{num2}" # The first number modulus the second number is invalid, as 1 could not perform modulus by 0
-  end
-
-  return num1.to_f % num2.to_f
+  num1.to_f % num2.to_f
 end
 
 puts "Simple Calculator"
 20.times { print "-"}
 puts 
 
+# Declare Variable
+$num_1 
+$num_2
+
 #1
-puts "Please enter the first number" 
-num_1 = gets.chomp
-puts "Please enter the second number"
-num_2 = gets.chomp
+loop do
+  puts "Please enter the first number" 
+  $num_1 = gets.chomp
+
+  puts "Please enter the second number"
+  $num_2 = gets.chomp
+
+  begin
+    # To check if the input is Number
+    Float($num_1)
+    Float($num_2)
```

Comment:
Why not just do:
```suggestion
    $num_1 = Float($num_1)
    $num_2 = Float($num_2)
```

Reply:
I thought of showing this error message here of what user input like they enter `10`: 
`"Invalid for second number, as #{$num_2} could not perform division or modulus operation."`
show `10` instead of `10.0`

Reply:
Can you try input 

"a" as num_1 and tell me if the error shown make sense or not

Reply:
> Can you try input
> 
> "a" as num_1 and tell me if the error shown make sense or not


For like String input I use exception when try to use `Float(num_1)` and show this error

```
  rescue ArgumentError => e
    puts "\nInput must be Number"
    puts "Please try again.\n\n"
  end
```

Reply:
>I thought of showing this error message here of what user input like they enter 10:
"Invalid for second number, as #{$num_2} could not perform division or modulus operation."
show 10 instead of 10.0
Ok. I don't understand your this as a reply to my comment

Reply:
> Ok. I don't understand your this as a reply to my comment

I mean showing the value `0` when user input `0` instead of `0.0` in this message:
`"Invalid for second number, as 0 could not perform division or modulus operation."`
But I think after the calculation all also become float anyway ahaha